Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt

```diff
@@ -0,0 +1,333 @@
+            is AnnotationStub.ObjC.Method -> TODO()
+            is AnnotationStub.ObjC.Factory -> TODO()
+            AnnotationStub.ObjC.Consumed -> TODO()
+            is AnnotationStub.ObjC.Constructor -> TODO()
+            is AnnotationStub.ObjC.ExternalClass -> TODO()
+            AnnotationStub.CCall.CString -> TODO()
+            AnnotationStub.CCall.WCString -> TODO()
+            is AnnotationStub.CCall.Symbol ->
+                mapOf("id" to KmAnnotationArgument.StringValue(symbolName))
+            is AnnotationStub.CStruct -> TODO()
+            is AnnotationStub.CNaturalStruct -> TODO()
+            is AnnotationStub.CLength -> TODO()
+            is AnnotationStub.Deprecated -> TODO()
+        }
+        return KmAnnotation(classifier.fqNameSerialized, args)
+    }
+
+    fun StubType.map(): KmType {
+        return when (this) {
+            is ClassifierStubType -> KmType(flags).also { km ->
+                km.arguments += typeArguments.map { it.map() }
+                if (isTypealias) {
+                    km.abbreviatedType = abbreviatedType
+                    km.classifier = expandedType.map().classifier
+                } else {
+                    km.classifier = KmClassifier.Class(classifier.fqNameSerialized)
+                }
+
+            }
+            is FunctionalType -> KmType(flags).also { km ->
+                km.classifier = KmClassifier.Class(classifier.fqNameSerialized)
+            }
+            is TypeParameterType -> KmType(flags).also { km ->
+                km.classifier = KmClassifier.TypeParameter(id)
+            }
+        }
+    }
+
+    private val ClassifierStubType.abbreviatedType: KmType
+        get() = KmType(flags).also { km ->
+            km.classifier = KmClassifier.TypeAlias(classifier.fqNameSerialized)
+        }
+
+    fun FunctionParameterStub.map(): KmValueParameter =
+            KmValueParameter(flags, name).also { km ->
+                type.map().let {
+                    if (isVararg) {
+                        km.varargElementType = it
+                    } else {
+                        km.type = it
+                    }
+                }
+            }
+
+    fun TypeParameterStub.map(): KmTypeParameter =
+            KmTypeParameter(flagsOf(), name, id, KmVariance.INVARIANT).also { km ->
+                km.upperBounds.addIfNotNull(upperBound?.map())
+            }
+
+    private fun TypeArgument.map(): KmTypeProjection = when (this) {
+        TypeArgument.StarProjection -> KmTypeProjection.STAR
+        is TypeArgumentStub -> KmTypeProjection(variance.map(), type.map())
+        else -> error("Unexpected TypeArgument: $this")
+    }
+
+    private fun TypeArgument.Variance.map(): KmVariance = when (this) {
+        TypeArgument.Variance.INVARIANT -> KmVariance.INVARIANT
+        TypeArgument.Variance.IN -> KmVariance.IN
+        TypeArgument.Variance.OUT -> KmVariance.OUT
+    }
+
+    fun ConstantStub.map(): KmAnnotationArgument<*> = when (this) {
```

Comment:
The name of this function doesn't fit to its meaning.
Maybe `mapToAnnotationArgument`?